starter-project-backend-g32/src/controllers/user.Controller.ts

```diff
@@ -0,0 +1,166 @@
+import {Request, Response , NextFunction} from "express"
+import {User} from "../models/User.model";
+import bcrypt from "bcrypt";
+import { signJWT } from "../functions/signJWT";
+
+
+const saltRounds  = 10;
+ 
+export const getUsers = async function(req : Request , res : Response) {
+
+    try {
+        const users = await User.find();
+        res.json(users);
+    }catch (e) {
+        res.status(404)
+    }
+};
+
+export const getUserByUsername = async function(req: Request , res : Response) {
+    
+    try {  
+        const user = await User.findOne({username : req.params.username})
+        res.status(200).json(user)
+    } catch (err) {
+        res.status(400).json(err)
```

Comment:
we shouldn't send the whole err object to frontend. It's better to send err.message